subsys/dfu/dfu_target/src/dfu_target_mcuboot.c

```diff
@@ -187,11 +216,28 @@ int dfu_target_mcuboot_done(bool successful)
 	if (successful) {
 		stream_buf_bytes = 0;
 
-		err = stream_flash_erase_page(dfu_target_stream_get_stream(),
-					secondary_last_address[curr_sec_img]);
-		if (err != 0) {
-			LOG_ERR("Unable to delete last page: %d", err);
-			return err;
+		int way = _chose_done_way();
+
+		if (way == _DO_ERASE) {
+#ifdef CONFIG_STREAM_FLASH_ERASE
+			err = stream_flash_erase_page(dfu_target_stream_get_stream(),
+						secondary_last_address[curr_sec_img]);
+			if (err != 0) {
+				LOG_ERR("Unable to delete last page: %d", err);
+				return err;
+			}
+#endif
+		} else {
+			/* instead of the erase of the trailer image page, clear the trailer status chunk */
+			ssize_t status_offset = boot_get_trailer_status_offset(secondary_size[cur_sec_img]);
+			size_t offset;
+			err = dfu_target_stream_offset_get(&offset);
+
+			if (err == 0) && (offset < secondary_size[cur_sec_img])
+			{
+			err = flash_flatten(secondary_dev[cur_sec_img], status_offset,
+							   secondary_size[cur_sec_img] - status_offset);
+			}
```

Comment:
Consider scrambling trailer just with writing improper magic word.

Reply:
I need to check whether MCUboot will work with that, but it should. Benefit of doing the least write is that erase/flatten starts from the beginning of device and goes forward and has to erase/write several bytes; if you can make data invalid by changing the single byte, we can just do that.